src/edu/washington/escience/myria/coordinator/catalog/MasterCatalog.java

```diff
@@ -1753,6 +1753,40 @@ private void deleteRelationIfExists(@Nonnull final SQLiteConnection sqliteConnec
   }
 
   /**
+   * Connects to SQLite to delete
+   * 
+   * @throws CatalogException if there is an error in the catalog.
+   */
+  public void deleteRelationFromCatalog(final RelationKey relationkey) throws CatalogException {
+    if (isClosed) {
+      throw new CatalogException("Catalog is closed.");
+    }
+    /* Do the work */
+    try {
+      queue.execute(new SQLiteJob<Void>() {
+        @Override
+        protected Void job(final SQLiteConnection sqliteConnection) throws CatalogException, SQLiteException {
+          sqliteConnection.exec("BEGIN TRANSACTION;");
+          try {
+            deleteRelationIfExists(sqliteConnection, relationkey);
+            sqliteConnection.exec("COMMIT TRANSACTION;");
+          } catch (SQLiteException e) {
+            try {
+              sqliteConnection.exec("ROLLBACK TRANSACTION;");
+            } catch (SQLiteException e2) {
+              assert true; /* Do nothing */
+            }
+            throw e;
+          }
+          return null;
+        }
+      }).get();
+    } catch (InterruptedException | ExecutionException e) {
```

Comment:
I don't think it ever makes sense to wrap an `InterruptedException`. Either put it in your `throws` clause (which you can't do if you're implementing `Runnable` but you can do with `Callable`), or call `Thread.currentThread().interrupt()` and swallow the exception. Either approach ensures that the caller (possibly scheduling code which you don't control) is properly notified of the interruption and can handle it appropriately (e.g., cancelling a `Future`). For example, perhaps someone wants to call *your* code in a `Future`--if you wrap `InterruptedException` they would see an interruption as an `ExecutionException` rather than an `InterruptedException`.